### PR TITLE
Catch failures; Fix simulation mode tests

### PR DESCRIPTION
**Pull request by CyanDevs**
This will do the following:

1) Failures when building with the bat script will now be caught by Jenkins as failures
2) In simulation mode, there are samples that should be skipped. See items appended to SAMPLES_LIST under the IF statement with OE_SIMULATION in https://github.com/openenclave/openenclave/blob/master/samples/test-samples.cmake#L54
3) In simulation mode, samples should use target `simulate` instead of `run` when available.

Signed-off-by: Chris Yan <chrisyan@microsoft.com>

### CODE REVIEW

File: src/jenkins/common/Openenclave.groovy
Comment (CyanDevs) [inline src/jenkins/common/Openenclave.groovy]: file-encryptor is skipped but also has a `simulate` target in the Makefile. Right now this PR has it running ninja simulate and tests have passed: https://oe-jenkins-dev.westeurope.cloudapp.azure.com/blue/organizations/jenkins/pipelines%2FAzure-Windows/detail/Azure-Windows/3035/pipeline/14/

Do we want to skip or use `ninja simulate`?

Comment (radhikaj) [reply src/jenkins/common/Openenclave.groovy]: If it works, we should run ninja simulate.

**Review by radhikaj — APPROVED** [review]

**Review by rs-- — APPROVED** [review]

**Comment by ghost** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/openenclave/openenclave-ci/pull/31#" title="Author self-approved">CyanDevs</a>*, *<a href="https://github.com/openenclave/openenclave-ci/pull/31#pullrequestreview-688839345" title="Approved">radhikaj</a>*, *<a href="https://github.com/openenclave/openenclave-ci/pull/31#pullrequestreview-688840503" title="Approved">rs--</a>*

The full list of commands accepted by this bot can be found [here](?repo=openenclave%2Fopenenclave-ci).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/openenclave/openenclave-ci/blob/master/OWNERS)~~ [rs--]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ghost** [comment]
@CyanDevs: you cannot LGTM your own PR.

<details>

In response to [this](https://github.com/openenclave/openenclave-ci/pull/31#issuecomment-865329479):

>/lgtm


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Pull request merged**
